src/linear_eq.jl

```diff
@@ -0,0 +1,217 @@
+"""
+Preconditions the matrix A and b with the inverse of mid(A)
+"""
+function preconditioner(A::AbstractMatrix, b::AbstractArray)
+
+    Aᶜ = mid.(A)
+    B = inv(Aᶜ)
+
+    return B*A, B*b
+
+end
+
+function gauss_seidel_interval(A::AbstractMatrix, b::AbstractArray; precondition=true, maxiter=100)
+
+    n = size(A, 1)
+    x = fill(-1e16..1e16, n)
+
+    if (typeof(b) <: SArray || typeof(b) <: MArray)
+        x = MVector{n}(x)
```

Comment:
@dpsanders According to the julia docs (https://docs.julialang.org/en/stable/manual/performance-tips/#Avoid-changing-the-type-of-a-variable-1), explicit conversions can be used to avoid optimization loss. Can `x` be changed to `MVector{n}` in this way?

Reply:
I have added the new benchmarks as well, they seem similar to the earlier ones.

Reply:
No, you should not do this in the case of an `SArray` because it changes the type of `x`, thereby introducing a type instability. (I think you misread the Julia docs, although maybe they aren't very clear.)

Reply:
Okay, got it.
What do the docs mean when they say
> There are several possible fixes:
>  Use an explicit conversion: `x = oneunit(T)`